Code/GraphMol/Atom.cpp

```diff
@@ -166,15 +166,10 @@ unsigned int Atom::getTotalNumHs(bool includeNeighbors) const {
                "valence not defined for atoms not associated with molecules")
   int res = getNumExplicitHs() + getNumImplicitHs();
   if (includeNeighbors) {
-    ROMol::ADJ_ITER begin, end;
-    const ROMol *parent = &getOwningMol();
-    boost::tie(begin, end) = parent->getAtomNeighbors(this);
-    while (begin != end) {
-      const Atom *at = parent->getAtomWithIdx(*begin);
-      if (at->getAtomicNum() == 1) {
-        res++;
+    for (auto nbr : getOwningMol().atomNeighbors(this)) {
```

Comment:
I know you don't want to modify the c++ api too much, but I'll give it one push and  leave this pretty snippet here :)


```
for (auto nbr: atom->getNbrs()) {
}
```

I also think this is more discoverable with code completion :)

Reply:
I definitely see the appeal.
A concrete suggestion: how about we merge this as is and then figure out if there's a good way to add something equivalently simple. This could easily be part of the v2 API discussion.

Reply:
you know... the more I think about it, the more my position on this moderates...
maybe I'll do a PR with something this weekend.